prov/rxm/src/rxm.h

```diff
@@ -280,18 +281,26 @@ struct rxm_mr {
 	struct rxm_domain *domain;
 };
 
-struct rxm_ep_wire_proto {
-	uint8_t	ctrl_version;
-	uint8_t	op_version;
-	uint8_t endianness;
-	uint8_t padding[5];
-	uint64_t eager_size;
-};
-
-struct rxm_cm_data {
-	struct sockaddr name;
-	uint64_t conn_id;
-	struct rxm_ep_wire_proto proto;
+union rxm_cm_data {
+	/* connect */
+	struct {
+		uint8_t version;
+		uint8_t endianness;
+		uint8_t ctrl_version;
+		uint8_t op_version;
+		uint16_t port;
+		uint8_t padding[2];
+		uint64_t eager_size;
+		uint64_t client_conn_id;
+	};
+	/* accept */
+	struct {
+		uint64_t server_conn_id;
+	};
+	/* reject */
+	struct {
+		enum rxm_cmap_reject_flag reject_flag;
+	};
```

Comment:
Please define 3 different structures, rather than embedding anonymous structures in the union.  Replace the enum with a fixed sized value, like u8 or u32